applets/wncklet/window-list.c

```diff
@@ -237,16 +237,18 @@ static void applet_change_background(MatePanelApplet* applet, MatePanelAppletBac
 
 #ifdef HAVE_X11
 #ifdef HAVE_WINDOW_PREVIEWS
-static GdkPixbuf *preview_window_thumbnail (WnckWindow *wnck_window, TasklistData *tasklist)
+static cairo_surface_t *preview_window_thumbnail (WnckWindow *wnck_window, TasklistData *tasklist)
 {
 	GdkWindow *window;
-	GdkPixbuf *screenshot;
-	GdkPixbuf *thumbnail;
+	GdkDisplay *display;
+	cairo_surface_t *screenshot;
+	cairo_surface_t *thumbnail;
+	cairo_t *cr;
 	double ratio;
-	int width, height;
-	int scale;
+	int width, height, scale;
 
-	window = gdk_x11_window_foreign_new_for_display (gdk_display_get_default (), wnck_window_get_xid (wnck_window));
+	display = gdk_display_get_default ();
+	window = gdk_x11_window_foreign_new_for_display (display, wnck_window_get_xid (wnck_window));
```

Comment:
There is need to add a new display variable called display.

Reply:
it's there. But realized that this change is not really needed, so I removed it